docs/source/pagination/introduction.mdx

```diff
@@ -0,0 +1,127 @@
+
+The `previousPages` field contains all the pages that have been fetched in the reverse direction, and exposes the paginated query's `Data` type. The `initialPage` field contains the first page of data that was fetched, and is an instance of the initial query's `Data` type. The `nextPages` field contains all the pages that have been fetched in the forward direction, and is a list of the paginated query's `Data` type.
+
+If you want to use a custom response type, you can provide a `transform` closure when initializing the `GraphQLQueryPager` instance. This closure is called whenever the pager fetches a new page of data, and is used to transform the raw response into the desired response type. 
+
+```swift
+struct Person {
+    let id: String
+    let name: String
+}
+
+let pager = GraphQLQueryPager.makeForwardCursorQueryPager(
+    client: apolloClient,
+    queryProvider: { page in
+        MyQuery(first: 10, after: page?.cursor ?? .none)
+    },
+    extractPageInfo: { data in
+        CursorBasedPagination.Forward(
+            hasNext: data.values.pageInfo.hasNextPage ?? false,
+            endCursor: data.values.pageInfo.endCursor
+        )
+    },
+    transform: { data in 
+        data.values.nodes.map { Person(id: $0.id, name: $0.name) }
+    }
+)
+```
+
+In this example, the `GraphQLQueryPager` instance is initialized with a `transform` closure that transforms the raw response into an array of `Person` objects. Alternatively, we can make use of the `GraphQLQueryPager`'s `map` method to achieve the same result:
+
+```swift
+pager.map { data in 
+    data.values.nodes.map { Person(id: $0.id, name: $0.name) }
+}.sink { result in
+    switch result {
+    case .success(let people):
+        // Handle the people
+    case .failure(let error):
+        // Handle the error
+    }
+}
+```
+
+## Fetching Data
+
+The `GraphQLQueryPager` class provides several methods to fetch paginated data: `fetch`, `refetch`, `loadNext`, `loadPrevious`, and `loadAll`. 
+
+The `fetch` method is used to fetch the first page of data and must be called before `loadNext` or `loadPrevious` can be called. The `refetch` method is used to refetch the first page of data, and additionally resets the pager to its initial state. The `loadNext` method is used to fetch the next page of data, and the `loadPrevious` method is used to fetch the previous page of data. The `loadAll` method is used to fetch all pages of data. 
+
+## Cancelling a Pager
+
+The `GraphQLQueryPager` class provides a `cancel` method, which can be used to cancel any ongoing fetch operations as well as cease any watcher subscriptions. Once cancelled, the pager's state is reset. Any subscriber that was subscribed to the pager will remain subscribed and will continue to receive updates. 
```

Comment:
> Any subscriber that was subscribed to the pager will remain subscribed and will continue to receive updates.

If the watchers have been canceled, what updates would they receive. This is unclear to me.

I wonder if on `cancel`, we should send a termination signal to the subscribers?

Reply:
Hmmm I see. In FRP, I think the most common pattern is that a `cancel()` should terminate the publisher and dispose of all subscriptions. I'm concerned that deviating from that might cause confusion.

For the case of `refetch`, we don't want to do that of course, but in my mind, that means `refetch` should be calling a private function that cancels the internal network calls and existing watchers without terminating the publisher stream. `refetch` is managing some internal state to begin a refetch, but the subscribers will continue to get events because the pager itself isn't cancelled. While `cancel` should actually cancel the pager.

Especially considering the pager itself actually conforms to `Publisher`, I think `cancel` should do what you expect cancelling an FRP stream would do.

Reply:
Hmm...

I wonder how we would want to disambiguate this, and whether this is worth doing prior to first public release. What are your thoughts here?

Reply:
I mean, this is an 0.1 release, so we are allowed to make breaking changes. But this is a pretty significantly breaking change...

I don't think it would be too hard to disambiguate though. Let's chat about this one.